src/nvim/ops.c

```diff
@@ -3409,23 +3474,27 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
             old_pos = curwin->w_cursor;
             curwin->w_cursor.lnum = lnum;
             ptr = ml_get(lnum);
-            if (cnt == count && i == y_size - 1)
+            if (cnt == count && i == y_size - 1) {
               lendiff = (int)STRLEN(ptr);
-            if (*ptr == '#' && preprocs_left())
-              indent = 0;                   /* Leave # lines at start */
-            else if (*ptr == NUL)
-              indent = 0;                   /* Ignore empty lines */
-            else if (first_indent) {
+            }
+            if (*ptr == '#' && preprocs_left()) {
+              indent = 0;                   // Leave # lines at start
+            } else if (*ptr ==
```

Comment:
something went wrong here? :)

Reply:
I've added this fix as a separate commit. Take a look and see if it looks alright.

Reply:
Uncrustify is better at _not_ splitting lines than joining them. I think the simplest fix would be to adjust this manually and uncrustify won't touch it afterwards. Is that alright solution?

Reply:
Sure let's be conservative and do it manually if uncrustify cannot be taught the right thing

Reply:
:+1: